tests/osgi-tests/src/test/java/org/wso2/carbon/identity/mgt/test/osgi/PasswordPolicyPatternTests.java

```diff
@@ -0,0 +1,250 @@
+import org.wso2.carbon.identity.mgt.claim.Claim;
+import org.wso2.carbon.identity.mgt.exception.AuthenticationFailure;
+import org.wso2.carbon.identity.mgt.exception.IdentityStoreException;
+import org.wso2.carbon.identity.mgt.exception.UserNotFoundException;
+import org.wso2.carbon.identity.mgt.test.osgi.util.IdentityMgtOSGITestConstants;
+import org.wso2.carbon.identity.mgt.test.osgi.util.IdentityMgtOSGiTestUtils;
+import org.wso2.carbon.kernel.utils.CarbonServerInfo;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.inject.Inject;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.PasswordCallback;
+
+import static org.ops4j.pax.exam.CoreOptions.systemProperty;
+
+/**
+ * Validate user given password according to predefine conditions such as min length, max length
+ * todo: Need to consider below test cases for add user operation as well
+ */
+@Listeners(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+public class PasswordPolicyPatternTests {
+
+    private static List<User> users = new ArrayList<>();
+    private static final Logger LOGGER = LoggerFactory.getLogger(PasswordPolicyPatternTests.class);
+    private static final String PASSWORD_PREFIX = "admin";
+
+    @Inject
+    private BundleContext bundleContext;
+
+    @Inject
+    private CarbonServerInfo carbonServerInfo;
+
+    @Configuration
+    public Option[] createConfiguration() {
+
+        List<Option> optionList = IdentityMgtOSGiTestUtils.getDefaultSecurityPAXOptions();
+
+        optionList.add(systemProperty(IdentityMgtOSGITestConstants.JAVA_SEC_SYSTEM_PROPERTY)
+                .value(Paths.get(IdentityMgtOSGiTestUtils.getCarbonHome(),
+                        IdentityMgtOSGITestConstants.CARBON_DIRECTORY_CONF,
+                        IdentityMgtOSGITestConstants.CARBON_DIRECTORY_SECURITY,
+                        IdentityMgtOSGITestConstants.JAAS_CONFIG_FILE).toString()));
+
+        return optionList.toArray(new Option[optionList.size()]);
+    }
+
+    @Test(groups = "passwordPolicyPatternValidation", description = "Password Policy Pattern Validation without " +
+            "special characters")
+    public void passwordPatternValidationWithoutSpecialCharacters() throws IdentityStoreException,
+            AuthenticationFailure, UserNotFoundException {
+
+        String uniqueUserId = addUser("testUser1");
+        RealmService realmService = null;
+        try {
+            realmService = bundleContext.getService(bundleContext.getServiceReference(RealmService.class));
+            Assert.assertNotNull(realmService, "Failed to get realm service instance");
+
+            try {
+                realmService.getIdentityStore().
+                        updateUserCredentials(uniqueUserId, generateCredentials(PASSWORD_PREFIX + "A1"));
+            } catch (Exception e) {
+                Assert.assertTrue((e instanceof IdentityStoreException));
```

Comment:
In all the occasions where 'Assert.assertTrue' is used, please provide the message for failure case.